gestalt-asset-core/src/main/java/org/terasology/gestalt/assets/AssetType.java

```diff
@@ -75,8 +75,7 @@
     private final Class<U> assetDataClass;
     private final AssetFactory<T, U> factory;
     private final List<AssetDataProducer<U>> producers = Lists.newCopyOnWriteArrayList();
-    private final Map<ResourceUrn, T> loadedAssets = new MapMaker().concurrencyLevel(4).makeMap();
-    private final ListMultimap<ResourceUrn, WeakReference<T>> instanceAssets = Multimaps.synchronizedListMultimap(ArrayListMultimap.<ResourceUrn, WeakReference<T>>create());
```

Comment:
I don't understand why the fix for the leakage isn't just iterating through the values of the resource urn and cleaning up the null references on disposal rather than this... complex manual linked-list implementation.